src/libopensc/sc.c

```diff
@@ -72,6 +72,13 @@ int sc_hex_to_bin(const char *in, u8 *out, size_t *outlen)
 	int r = SC_SUCCESS;
 	size_t left = *outlen;
 	u8 byte = 0;
+	/* skip initial separators */
```

Comment:
Initial seperators should be covered in the loop. In the first run byte_needs_nibble is 0 so that the loop will be continued in line 97, won't it?